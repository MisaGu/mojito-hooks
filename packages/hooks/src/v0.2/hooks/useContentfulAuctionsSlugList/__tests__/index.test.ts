import { act, renderHook } from '@testing-library/react-hooks';
import { TestWrapper } from '../../../components/test/wrapper/TestWrapper';
import {
  useContentfulAuctionsSlugList,
  UseContentfulAuctionsSlugListProps,
  UseContentfulAuctionsSlugListReturn,
} from '../useContentfulAuctionsSlugList';

describe('useContentfulAuctionsSlugList()', () => {
  it('should be defined', () => {
    expect(useContentfulAuctionsSlugList).toBeDefined();
  });

  it('returns the right result when needed', async () => {
    const { result, rerender, waitFor } = renderHook<
      [UseContentfulAuctionsSlugListProps?],
      UseContentfulAuctionsSlugListReturn
    >(
      (args = []) => {
        return useContentfulAuctionsSlugList(...args);
      },
      {
        wrapper: TestWrapper as any,
      },
    );

    // Trigger request...
    await act(async () => rerender());

    // ...and wait for response:
    await waitFor(() => result.current.queryResult.isSuccess);

    // Check data matches snapshot:
    expect(result.current).toMatchSnapshot({
      queryResult: {
        dataUpdatedAt: expect.any(Number),
      },
    });
  });
});
