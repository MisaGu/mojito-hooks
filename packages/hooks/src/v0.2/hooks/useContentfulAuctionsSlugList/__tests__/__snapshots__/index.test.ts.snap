// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useContentfulAuctionsSlugList() returns the right result when needed 1`] = `
Object {
  "auctionsSlugList": Array [
    "mock-item-1",
    "mock-item-2",
  ],
  "error": null,
  "isLoading": false,
  "queryResult": Object {
    "dataUpdatedAt": Any<Number>,
    "errorUpdateCount": 0,
    "errorUpdatedAt": 0,
    "failureCount": 0,
    "isError": false,
    "isFetched": true,
    "isFetchedAfterMount": true,
    "isFetching": false,
    "isIdle": false,
    "isLoadingError": false,
    "isPlaceholderData": false,
    "isPreviousData": false,
    "isRefetchError": false,
    "isRefetching": false,
    "isStale": false,
    "isSuccess": true,
    "remove": [Function],
    "status": "success",
  },
  "refetch": [Function],
}
`;
